kernel/src/hil/framebuffer.rs

```diff
@@ -0,0 +1,57 @@
+//! Hil for FrameBuffer
+use crate::returncode::ReturnCode;
+use enum_primitive::cast::FromPrimitive;
+use enum_primitive::enum_from_primitive;
+
+enum_from_primitive! {
+    #[derive(Copy, Clone, PartialEq)]
+    pub enum ScreenRotation {
+        Normal = 0,
+        Rotated90 = 1,
+        Rotated180 = 2,
+        Rotated270 = 3,
+    }
+}
+
+impl From<ScreenRotation> for usize {
+    fn from(rotation: ScreenRotation) -> usize {
+        match rotation {
+            ScreenRotation::Normal => 0,
+            ScreenRotation::Rotated90 => 1,
+            ScreenRotation::Rotated180 => 2,
+            ScreenRotation::Rotated270 => 3,
+        }
+    }
+}
+
+pub trait Screen {
+    fn set_resolution(&self, width: usize, height: usize) -> ReturnCode;
+    fn set_color_depth(&self, depth: usize) -> ReturnCode;
+    fn set_rotation(&self, rotation: ScreenRotation) -> ReturnCode;
```

Comment:
I wonder if these would make it too difficult to implement this HIL, if the screen does not somehow support these operations in hardware. I would think we don't need to support changing the resolution, etc, but these could be in an "Advanced" HIL.

Reply:
I was not sure how to handle this. First I also thought about two HILs, what is not clear to me is how these would interact with the frame buffer capsule? I think there might be two options:
  - have another advanced frame buffer capsule, actually a layer on top of frame buffer
  - add a default implementation for these functions, reporting SUCCESS if the resolution and color depth fits the one implemented by the display or ENOSUPORT otherwise.

I think ``set_rotation`` is available on most displays, as it actually changes only the way the bytes are written to the memory.

Reply:
We have the pattern of layered advanced HILs in Uart currently, though I don't think any implementation of this idiom has ever landed in upstream Tock. The idea is that a capsule can implement both the basic Uart HIL as well as an Advanced HIL, eg: https://github.com/lab11/signpost-software/blob/master/signpost/kernel/signpost_drivers/src/gps_console.rs#L39

Reply:
The SAM4L implements the advanced UART, and the tock bootloader uses it.

Reply:
I added a ScreenSetup trait and an Option for it to the framebuffer capsule.